src/Compilers/CSharp/Portable/Syntax/SyntaxKindFacts.cs

```diff
@@ -8,7 +8,9 @@ public static partial class SyntaxFacts
     {
         public static bool IsKeywordKind(SyntaxKind kind)
         {
-            return IsReservedKeyword(kind) || IsContextualKeyword(kind);
+            return IsReservedKeyword(kind)
+                || IsContextualKeyword(kind)
+                || kind == SyntaxKind.DiscardDesignation;
```

Comment:
@SpencerLynn You could revert SyntaxKindFacts and do the DiscardDesignation check in ClassificationHelpers.GetClassification

Reply:
Sounds good. We can definitely put this back in `ClassificationHelpers.GetClassification`.

@CyrusNajmabadi, do you know if there is a better place for it? Does it fall under `IsContextualKeyword` perhaps? If you think it isn't a syntax fact at all, then perhaps @JoeRobich's suggestion is the best place for it.

Reply:
It would be neither.  Both the 'contextual' and 'regular' keywords end with 'Keyword'.   In thsi case, it's just a DicardDesignation.  It's just that we've chosen to *color* it the special "this is an important language word" color.  So this logic belongs in teh classifier.

Reply:
You'll notice this is why 'dynamic' and 'var' are also not considered kewords.  They're not actually keywords.  They're just special language constructs.  We just choose to color them as keywords in teh classifier because we present a simpler view to users.  However, we want our internal APIs to be correct wrt to the actual language here since it can be super confusing otherwise.

Reply:
Sounds good. Thanks for the information!

Reply:
`var` is a contextual keyword in C# 8.  The pattern `var (x, y)` introduces two variables named `x` and `y`, while the pattern `Tuple(X, Y)` checks that the input is of type Tuple, and checks that its two elements are equal to the in-scope constants named `X` and `Y`.  They are parsed differently.

Reply:
Similarly, the `_` identifier is now a contextual keyword, at least in patterns.  It is parsed into a different syntax node than an identifier would be.

Reply:
@gafter The IDE helpers here respect the underlying SyntaxFacts.  So if '_' or 'var' are reported as contxtual keywords through SyntaxFacts.IsContextualKeyword, then we will as well.